Notification about not empty category while deleting

tungyr-one opened the pull request:
Let user know what he can lost

Commit:
More or less working

coder1coder commented on DocumentApp/ClientApp/src/app/doc-list/doc-list.component.ts:
```diff
@@ -60,7 +60,11 @@ export class DocumentListComponent implements OnInit
-      }
+      },
+      error: error => {
+        console.log(error);
```
убрать console.log

tungyr-one replied:
My bad

coder1coder commented on DocumentApp/api/Exceptions/ValidationsException.cs:
```diff
@@ -0,0 +1,21 @@
+
+namespace api.Exceptions
+{
+    public class ValidationException: ArgumentException
```
Можно унаследовать просто от Exception

tungyr-one replied:
Точно

coder1coder commented on DocumentApp/api/Services/CategoriesService.cs:
```diff
@@ -14,10 +19,12 @@ public class CategoriesService : ICategoriesService
+      private readonly DataContext _context;
 
-        public CategoriesService(ICategoriesRepository categoryRepository, ILogger<CategoriesService> logger,
+        public CategoriesService(ICategoriesRepository categoryRepository, DataContext context, ILogger<CategoriesService> logger,
```
прокидывать DbContext только в репозиториях

tungyr-one replied:
Запомню

coder1coder commented on DocumentApp/api/Services/CategoriesService.cs:
```diff
@@ -55,7 +62,19 @@ public async Task<bool> UpdateAsync(int id, CategoryDto categoryUpdate)
+                throw new ValidationException("Category has subcategories, delete them first");
+            }
+
+            if (await _context.Docs.AnyAsync(d => d.Category.Id == id))
```
Инкапсулировать в метод репозитория Exists

tungyr-one replied:
Добавил метод в последнем коммите:
` async Task<bool> IDocsRepository.IsDocumentWithCategoryRelationExists(int categoryId)
        {
            return await _context.Docs.AnyAsync(d => d.Category.Id == categoryId);
        }
`

Commit:
Correct issues bf merge to master

Moved context to repo, made another method to check doc-category relation, cleaned small stuff

Commit:
Merge branch 'master' into categorySafelyRemoval
